### ISSUE TITLE
TestBackendLoadRetry fail

### ISSUE THREAD

**Issue by copart** [restic/restic#3033]
Trying to build restic 0.10.0 on Fedora Rawhide for packaging.

```
github.com/restic/restic/internal/backend
--- FAIL: TestBackendSaveRetry (0.03s)
    backend_retry_test.go:43: injected error
backend_retry_test.go:85: unexpected error: test list error
github.com/restic/restic/internal/backend.TestBackendListRetry.func1
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry_test.go:68
github.com/restic/restic/internal/mock.(*Backend).List
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/mock/backend.go:105
github.com/restic/restic/internal/backend.(*RetryBackend).List.func1
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry.go:127
github.com/cenkalti/backoff.RetryNotifyWithTimer
        /usr/share/gocode/src/github.com/cenkalti/backoff/retry.go:52
github.com/cenkalti/backoff.RetryNotify
        /usr/share/gocode/src/github.com/cenkalti/backoff/retry.go:31
github.com/restic/restic/internal/backend.(*RetryBackend).retry
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry.go:36
github.com/restic/restic/internal/backend.(*RetryBackend).List
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry.go:126
github.com/restic/restic/internal/backend.TestBackendListRetry
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry_test.go:81
testing.tRunner
        /usr/lib/golang/src/testing/testing.go:1123
runtime.goexit
        /usr/lib/golang/src/runtime/asm_amd64.s:1374

--- FAIL: TestBackendListRetry (0.00s)
backend_retry_test.go:246: unexpected error: reader reached limit of 100
github.com/restic/restic/internal/backend.failingReader.Read
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry_test.go:202
bytes.(*Buffer).ReadFrom
        /usr/lib/golang/src/bytes/buffer.go:204
io/ioutil.readAll
        /usr/lib/golang/src/io/ioutil/ioutil.go:36
io/ioutil.ReadAll
        /usr/lib/golang/src/io/ioutil/ioutil.go:45
github.com/restic/restic/internal/backend.TestBackendLoadRetry.func2
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry_test.go:243
github.com/restic/restic/internal/mock.(*Backend).Load
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/mock/backend.go:74
github.com/restic/restic/internal/backend.(*RetryBackend).Load.func1
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry.go:79
github.com/cenkalti/backoff.RetryNotifyWithTimer
        /usr/share/gocode/src/github.com/cenkalti/backoff/retry.go:52
github.com/cenkalti/backoff.RetryNotify
        /usr/share/gocode/src/github.com/cenkalti/backoff/retry.go:31
github.com/restic/restic/internal/backend.(*RetryBackend).retry
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry.go:36
github.com/restic/restic/internal/backend.(*RetryBackend).Load
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry.go:77
github.com/restic/restic/internal/backend.TestBackendLoadRetry
        /builddir/build/BUILD/restic-0.10.0/_build/src/github.com/restic/restic/internal/backend/backend_retry_test.go:242
testing.tRunner
        /usr/lib/golang/src/testing/testing.go:1123
runtime.goexit
        /usr/lib/golang/src/runtime/asm_amd64.s:1374

--- FAIL: TestBackendLoadRetry (0.00s)
FAIL
exit status 1
FAIL    github.com/restic/restic/internal/backend       2.259s
```

Expected behavior
-----------------
Build and pass all tests

Actual behavior
---------------
Fails the following tests TestBackendSaveRetry TestBackendListRetry TestBackendLoadRetry

Steps to reproduce the behavior
-------------------------------
fedpkg mockbuild

Do you have any idea what may have caused this?
-----------------------------------------------
Not yet

Do you have an idea how to solve the issue?
-------------------------------------------
Not yet

**Comment by greatroar** [comment]
Looks like the build is using github.com/cenkalti/backoff from an external source. If that's too new a version, the tests will fail. #2934 fixed this post-0.10.0.

**Comment by MichaelEischer** [comment]
Closing this as the issue has already been fixed in #2934. Just backport that change, is doesn't depend on anything else. It's not unexpected that the build or tests fail when using a dependency with a different _major_ version.

**Comment by copart** [comment]
Thank you @greatroar
